### ISSUE TITLE
jaccard and other qualitative methods not working as expected

### ISSUE THREAD

**Issue by gregcaporaso** [qiime2/q2-diversity#189]
A serious bug has been identified in the Jaccard distance metric, as well as several other qualitative diversity metrics imported from scikit-bio and SciPy. Unweighted UniFrac is **not** impacted by this bug. This seems to have resulted from a misunderstanding on our part of the definition of those methods.

Next steps:
 - I am working on understanding exactly what metrics are impacted by this, and how we will solve the issue; I will post to the QIIME 2 Forum to announce this today, once I fully understand the issue. 
 - @ebolyen is working on adding functionality to q2view that will allow users to determine if their qza or qzv file is impacted by this bug (should be ready today). 
 - @thermokarst is getting us ready to patch 2018.2 (should be out this week). 

Thanks @mahermassoud and @wasade for identifying this issue.

**Comment by gregcaporaso** [comment]
This is a false alarm as far as QIIME 2 is concerned, as scikit-learn is handling the conversion to presence/absence data for us. Here's some code that illustrates that:

```python
>>> import qiime2
>>> import qiime2.plugins.feature_table
>>> import qiime2.plugins.diversity
>>> import skbio
>>> import biom

# moving pictures (2018.2) rarefied table
>>> table = qiime2.Artifact.load('./rarefied-table.qza')
# moving pictures (2018.2) rooted tree
>>> tree = qiime2.Artifact.load('./rooted-tree.qza')

# At the moment, the diversity metrics do not support receiving FeatureTable[PresenceAbsence] as input,
# so for the purpose of this analysis we'll read that back into a FeatureTable[Frequency]
>>> pa_table = qiime2.plugins.feature_table.actions.presence_absence(table).presence_absence_table
>>> pa_table = qiime2.Artifact.import_data('FeatureTable[Frequency]', pa_table.view(biom.Table))

# For each diversity metric, test whether computing on frequency data versus computing on 
# relative abundance data results in equal distance matrices. This should be True for qualitative 
# metrics and false for quantitative metrics. 
>>> phyl_metrics = {'unweighted_unifrac', 'weighted_unifrac'}
>>> for metric in phyl_metrics:
        dm1 = qiime2.plugins.diversity.actions.beta_phylogenetic(table, tree, metric=metric).distance_matrix
        dm2 = qiime2.plugins.diversity.actions.beta_phylogenetic(pa_table, tree, metric=metric).distance_matrix
        print(metric, dm1.view(skbio.DistanceMatrix) == dm2.view(skbio.DistanceMatrix))

weighted_unifrac False
unweighted_unifrac True

>>> nonphyl_metrics = {'braycurtis', 'canberra', 'chebyshev', 'cityblock', 'correlation', 'cosine', 
                       'dice', 'euclidean', 'hamming', 'jaccard', 'kulsinski', 'mahalanobis', 'matching', 
                       'rogerstanimoto', 'russellrao', 'seuclidean', 'sokalmichener', 'sokalsneath', 
                       'sqeuclidean', 'wminkowski', 'yule'}
>>> for metric in nonphyl_metrics:
        try:
            dm1 = qiime2.plugins.diversity.actions.beta(table, metric=metric).distance_matrix
            dm2 = qiime2.plugins.diversity.actions.beta(pa_table, metric=metric).distance_matrix
        except ValueError as e:
            print(metric, e)
        else:
            print(metric, dm1.view(skbio.DistanceMatrix) == dm2.view(skbio.DistanceMatrix))


mahalanobis The number of observations (33) is too small; the covariance matrix is singular. For observations with 675 dimensions, at least 676 observations are required.
hamming False
seuclidean False
cityblock False
chebyshev False
cosine False
sokalsneath True
yule True
jaccard True
sokalmichener True
kulsinski True
wminkowski weighted minkowski requires a weight vector `w` to be given.
correlation False
rogerstanimoto True
dice True
matching True
canberra False
sqeuclidean False
euclidean False
braycurtis False
russellrao True
```

The relevant code in q2-diversity is [here](https://github.com/qiime2/q2-diversity/blob/d3f39790565dfc898ea4e0ca6ca28172d51d1cb9/q2_diversity/_beta/_method.py#L69) and in sklearn [here](https://github.com/scikit-learn/scikit-learn/blob/a24c8b464d094d2c468a16ea9f8bf8d42d949f84/sklearn/metrics/pairwise.py#L1238).
